packages/mui-material/src/Rating/Rating.js

```diff
@@ -524,25 +550,54 @@ const Rating = React.forwardRef(function Rating(inProps, ref) {
 
   const classes = useUtilityClasses(ownerState);
 
+  const externalForwardedProps = {
+    slots,
+    slotProps,
+  };
+
+  const [RootSlot, rootSlotProps] = useSlot('root', {
+    ref: handleRef,
+    className: clsx(classes.root, className),
+    elementType: RatingRoot,
+    externalForwardedProps: {
+      ...externalForwardedProps,
+      ...other,
+      component,
+    },
+    getSlotProps: (handlers) => ({
+      ...handlers,
+      onMouseMove: (event) => {
+        handleMouseMove(event);
+        handlers.onMouseMove?.(event);
+      },
+      onMouseLeave: (event) => {
+        handleMouseLeave(event);
+        handlers.onMouseLeave?.(event);
+      },
+    }),
+    ownerState,
+    additionalProps: {
+      role: readOnly ? 'img' : null,
+      'aria-label': readOnly ? getLabelText(value) : null,
+    },
+  });
+
+  const [LabelSlot, labelSlotProps] = useSlot('label', {
+    className: clsx(classes.label, classes.labelEmptyValue),
+    elementType: RatingLabel,
+    externalForwardedProps,
+    ownerState,
+  });
+
+  const [DecimalSlot, decimalSlotProps] = useSlot('decimal', {
+    className: classes.decimal,
+    elementType: RatingDecimal,
+    externalForwardedProps,
+    ownerState,
+  });
+
   return (
-    <RatingRoot
-      as={component}
-      ref={handleRef}
-      onMouseMove={handleMouseMove}
-      onMouseLeave={handleMouseLeave}
-      className={clsx(
-        classes.root,
-        {
-          // TODO v6: remove this class as it duplicates with the global state class Mui-readOnly
-          'MuiRating-readOnly': readOnly,
-        },
```

Comment:
I agree with removing this, but we should add it to the upgrade guide: https://github.com/mui/material-ui/blob/master/docs/data/material/migration/upgrade-to-v7/upgrade-to-v7.md